### PR TITLE
Task Endpoint functionality implemented. Closes #28

### LINKED ISSUE #28: Tasks Root Endpoint Functionality
Task:
* `name`
* `description`
* `completed` (Boolean)
* `creator` (`Member`)
* `assignee` (`Member`)

Endpoints:
* `GET /tasks`
  * Filters: `creator, assignee, state`
* `POST /tasks`
  * Roles: `member`
* `GET /tasks/{id}`
* `PUT /tasks/{id}`
  * Roles: `member`
* `DELETE /tasks/{id}`
  * Roles: `officer`

### PR DESCRIPTION
**Pull request by kocsenc**

### CODE REVIEW

File: app/Http/Controllers/MemberController.php [hunk 1]
```diff
@@ -101,6 +101,7 @@ public function show($id)
     public function update(Request $request, $id)
     {
         $member = Member::findOrFail($id);
+        // TODO handle exception
```
Comment (craigcabrey) [inline app/Http/Controllers/MemberController.php]: Yea I'll fix that now.

File: app/Http/Controllers/TasksController.php [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+     * @param  int $id
+     * @return Response
+     */
+    public function destroy($id)
+    {
+        try {
+            Task::findOrFail($id);
+            Task::destroy($id);
+
+            return response('', Response::HTTP_NO_CONTENT);
```
Comment (craigcabrey) [inline app/Http/Controllers/TasksController.php]: Make this
`response(null, Response::HTTP_NO_CONTENT)`

I need to change the others, I know.

File: app/Http/routes.php [hunk 3]
```diff
@@ -32,6 +32,7 @@
                 'mentors_url' => route('api.v1.mentors.index'),
                 'officers_url' => route('api.v1.officers.index'),
                 'quotes_url' => route('api.v1.quotes.index'),
+                'tasks-url' => route('api.v1.tasks.index'),
```
Comment (craigcabrey) [inline app/Http/routes.php]: `_` not `-`

File: app/Http/Controllers/TasksController.php [hunk 4]
```diff
@@ -0,0 +1,122 @@
...
+     *
+     * @param  Request $request
+     * @return Response
+     */
+    public function store(Request $request)
+    {
+        $this->validate($request, [
+            'name' => 'required|string',
+            'description' => 'required|string',
+            'assignee' => 'exists:member,id'
```
Comment (craigcabrey) [inline app/Http/Controllers/TasksController.php]: `,` at the end

File: app/Task.php [hunk 5]
```diff
@@ -0,0 +1,96 @@
...
+        'assignee_url',
+        'url',
+    ];
+
+    /**
+     * @var array hidden attributes
+     */
+    protected $hidden = [
+        'creator',
+        'assignee'
```
Comment (craigcabrey) [inline app/Task.php]: `,` at the end

File: app/Task.php [hunk 6]
```diff
@@ -0,0 +1,96 @@
...
+
+    /**
+     * The attributes that are mass assignable.
+     *
+     * @var array
+     */
+    protected $fillable = [
+        'name',
+        'description',
+        'completed'
```
Comment (craigcabrey) [inline app/Task.php]: `,` at the end

File: app/Task.php [hunk 7]
```diff
@@ -0,0 +1,96 @@
...
+    ];
+
+    protected $casts = [
+        'completed' => 'boolean' // Always cast Task->completed as bool
```
Comment (craigcabrey) [inline app/Task.php]: `,` at the end

File: app/Http/Controllers/TasksController.php (hunk 2 above)
Comment (kocsenc) [inline app/Http/Controllers/TasksController.php]: Make new issue replace response for HTTP_NO_CONTENT

File: app/Http/routes.php (hunk 3 above)
Comment (kocsenc) [inline app/Http/routes.php]: good call

**Pull request merged**